pages/docs/reference/compatibility.md

```diff
@@ -0,0 +1,106 @@
+    - OV compiler may not accept NV binaries (e.g. those that exhibit newer language features or APIs)
+- **BC** = BCL & BCB
+- **EXP** - Experimental feature
+  - see below
+- **NO** - No compatibility guarantees
+  - we'll do our best to offer smooth migration, but can give no guarantees
+  - migration is planned individually for every incompatible subsystem 
+
+---
+
+\* No changes *modulo bugs* means that if an important bug is found (e.g. in the compiler diagnostics or elsewhere), a fix for it may introduce a breaking change, but we are always very careful with such changes.
+
+## Compatibility guarantees for Kotlin releases
+
+**Kotlin for JVM**:
+  - patch updates are fully compatible
+  - minor version updates are backwards compatible
+
+| JVM  | 1.0 | 1.0.X | 1.1 | 1.1.X | ... | 2.0 |
+|---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| **1.0** | - | C | BC | BC | ... | ? |
+| **1.0.X** | C | - | BC | BC | ... | ?
+| **1.1** | BC | BC | - | C | ... | ?
+| **1.1.X** | BC | BC | C | - | .... | ?
+| **...** | ... | ... | ... | ... | ... | ... |
+| **2.0** | ? | ? | ? | ? | ... | - 
+
+**Kotlin for JS**: starting with Kotlin 1.1, both patch and minor version updates provide backward compatibility for the language (BCL), but no BCB.  
+
+| JS  | 1.0.X | 1.1 | 1.1.X | ... | 2.0 |
+|---:|:---:|:---:|:---:|:---:|:---:|
+| **1.0.X** | - |  EXP | EXP | ... | EXP |
+| **1.1** | EXP |  - | BCL | ... | ?
+| **1.1.X** | EXP | BCL | - | .... | ?
+| **...** | ... | ... | ... | ... | ... | ... |
+| **2.0** | EXP | ? | ? | ? | ... | - 
+
+**Kotlin Scripts**: both patch and minor version updates provide backward compatibility for the language (BCL), but no BCB.
+
+## Experimental features
+
+Experimental features, such as coroutines in Kotlin 1.1, have exemption from the compatibility modes listed above. Such features require an opt-in to use without compiler warning. Experimental features are at least backwards compatible for patch version updates, but we do guarantee no compatibility for minor version updates (migration aids will be provided where possible).    
+
+| EXP  | 1.1 | 1.1.X | 1.2 | 1.2.X | 
+|---:|:---:|:---:|:---:|:---:|
+| **1.1** | - | BC | NO | NO  
+| **1.1.X** | BC | - | NO | NO
+| **1.2** | NO | NO | - | BC 
+| **1.2.X** | NO | NO | BC | - 
+
+## EAP builds
+
+We publish Early Access Preview (EAP) builds to special channels where early adopters from the community may try them out and give us feedback. Such builds provide no compatibility guarantees whatsoever (although we do our best to keep them reasonable compatible with releases and with each other). Quality expectations for such builds are also much lower than for releases. Beta builds also fall under this category.
+
+**IMPORTANT NOTE**: all binaries compiled by EAP builds for BCB-version (e.g. 1.1-eap-X) are **rejected by release builds of the compiler**. We don't want any code compiled by pre-release versions to be kept around after a stable version is released. 
```

Comment:
I think we should stress that only binaries compiled by EAP builds of pre-release versions of Kotlin are going to be rejected. Binaries compiled by EAP builds of patch updates are completely fine.